File: pari2e.c
#include <stdio.h>
int is_sparse(int **mat, int R, int C){
    int zeros=0;
    for(int i=0;i<R;i++) for(int j=0;j<C;j++) if(mat[i][j]==0) zeros++;
    return zeros > (R*C)/2;
}
int main(){
    int data[3][3]={{0,0,3},{0,5,0},{7,0,0}};
    int *mat[3]; for(int i=0;i<3;i++) mat[i]=data[i];
    printf("Is sparse? %s\n", is_sparse(mat,3,3) ? "True" : "False");
    return 0;
}

File: pari2e.py
def is_sparse(mat):
    rows=len(mat); cols=len(mat[0]) if rows else 0
    zeros=sum(1 for r in mat for x in r if x==0)
    return zeros > (rows*cols)/2

m = [[0,0,3],[0,5,0],[7,0,0]]
print("Is sparse?", is_sparse(m))

